src/v/cluster/members_backend.h

```diff
@@ -157,6 +171,7 @@ class members_backend {
 
     // replicas reallocations in progress
     std::vector<update_meta> _updates;
+    ss::circular_buffer<members_manager::node_update> _raft0_updates;
```

Comment:
What is the need to have another reconciliation loop rather than doing it inline with the node update? (like it was before)

Reply:
we want to process raft0 updates asynchronously to the loop responsible for partition reallocation. The reallocation process may take a long time. It would prevent nodes from joining raft0 and receiving data.